swc/defs.bzl

```diff
@@ -36,27 +31,50 @@ attribute.
 )
 
 def swc(name, srcs = None, args = [], data = [], output_dir = False, swcrc = None, source_maps = False, out_dir = None, root_dir = None, **kwargs):
-    """Execute the swc compiler
+    """Execute the SWC compiler
 
     Args:
-        name: A name for the target
-        srcs: source files, typically .ts files in the source tree
-        data: runtime dependencies to be propagated in the runfiles
-        output_dir: whether to produce a directory output rather than individual files
-        args: additional arguments to pass to swc cli, see https://swc.rs/docs/usage/cli
-        source_maps: If set, the --source-maps argument is passed to the swc cli with the value.
-          See https://swc.rs/docs/usage/cli#--source-maps--s
+        name: A name for this target
+
+        srcs: List of labels of TypeScript source files.
+
+        data: Files needed at runtime by binaries or tests that transitively depend on this target.
+            See https://bazel.build/reference/be/common-definitions#typical-attributes
+
+        output_dir: Whether to produce a directory output rather than individual files
+
+        args: Additional options to pass to SWC cli, see https://swc.rs/docs/usage/cli
+
+        source_maps: If set, the --source-maps argument is passed to the SWC cli with the value, see https://swc.rs/docs/usage/cli#--source-maps--s
           True/False are automaticaly converted to "true"/"false" string values the cli expects.
-        swcrc: label of a configuration file for swc, see https://swc.rs/docs/configuration/swcrc
-        out_dir: base directory for output files relative to the output directory for this package
-        root_dir: a subdirectory under the input package which should be consider the root directory of all the input files
-        **kwargs: additional named parameters like tags or visibility
+
+        swcrc: Label of a .swcrc configuration file for the SWC cli, see https://swc.rs/docs/configuration/swcrc
+            Instead of a label, you can pass a dictionary matching the JSON schema.
+
+        out_dir: The base directory for output files relative to the output directory for this package
+
+        root_dir: A subdirectory under the input package which should be consider the root directory of all the input files
+
+        **kwargs: passed through to underlying [`swc_compile`](#swc_compile), eg. `visibility`, `tags`
     """
     if srcs == None:
         srcs = native.glob(["**/*.ts", "**/*.tsx"])
     elif not types.is_list(srcs):
         fail("srcs must be a list, not a " + type(srcs))
 
+    if type(swcrc) == type(dict()):
+        swcrc.setdefault("sourceMaps", source_maps)
+
+        write_file(
+            name = "_gen_swcrc_%s" % name,
+            out = "swcrc_%s.json" % name,
```

Comment:
what about `%s_swcrc.json` so that in the output tree it is sorted by the target name?